### ISSUE TITLE
[BUG]  Different behavior input video vs. images

### ISSUE THREAD

**Issue by sergionegri** [CMU-Perceptual-Computing-Lab/openpose#897]
### Type of Issue
BUG
1) we get 3 calibrated and synchronized videos
2) we stack them horizontally as expected
3) To avoid issues on keyframes that might be introduced by ffmpeg first we do a video-to-frame

4a) we perform a frame-to-video on ffmpeg and launch OP on that with the following command:

```
./build/examples/openpose/openpose.bin
--video ~/.../Video.mp4
--3d_views 3
--3d
--3d_min_views 2
--number_people_max 1
--camera_parameter_folder ~/.../camera_parameter_folder/
--output_resolution 640x360
--display 0 --render_pose 0
--write_json INPUTvideo/
```

4b) in parallel we run OP on the images directly with the following command:

```
./build/examples/openpose/openpose.bin
--image_dir ~/.../folderImages/
--3d_views 3
--3d
--3d_min_views 2
--number_people_max 1
--camera_parameter_folder ~/.../camera_parameter_folder/
--output_resolution 640x360
--display 0 --render_pose 0
--write_json INPUTimages/
```

What we observe is that, in spite of having the same number of json files created, their content is different, **sometimes dramatically** (e.g. point detected in one case, not detected in another).

Here is a screenshot (left from images, right from video):
![image](https://user-images.githubusercontent.com/10323575/47572505-6aa07a80-d93b-11e8-9728-ea4bda6bfe2d.png)

Please have a look at some interesting points:

![image](https://user-images.githubusercontent.com/10323575/47572958-7c365200-d93c-11e8-8c90-ed0b6776fef0.png)

Usually the delta in percentage between the video version and the image version is relatively low, but we have some "interesting" deviations (some are highlighted) especially in 3d. We get to the point of missing a detection (yellow).

Now, we know that ffmpeg introduces some artifacts, but they should not be so big as to screw up the frame completely.

Is there any difference between the image processing and the video processing of OP?

Thank you for the amazing repo!

**Comment by gineshidalgo99** [comment]
No there is not. My bet is the quality lost when converting video to JPG.

Could you try converting the frames to PNG (which is lossless compression?). That would prove whether it is OP or the quality loss of JPG.

Thanks!

**Comment by sergionegri** [comment]
We actually use BMP, so definitely there is no loss. As I wrote the only difference I might think of is the one related to artifacts when the video is created but jeeez, I don't think a compression artifact might make a detection happen when it would not happen without it. I am puzzled.

**Comment by gineshidalgo99** [comment]
We use OpenCV to read both video and image. After that, it's a simple RGB image in cv::Mat format.There is no other difference in image/video other than that. Other than that, maybe you are undistorting the image only in one of the 2 cases or the OpenPose flags are not equivalent? Not sure.

**Comment by sergionegri** [comment]
Camera matrixes are identical, we never touch them, so I'd disregard different undistort behavior. 
Flags are also identical (we copypasted them above).

**Comment by gineshidalgo99** [comment]
Good observation. Let's split the problem and focus in the 2D keypoint part for now. Could i ask for a video (and the corresponding images) where it happens? I.e., where the 2D keypoints are different? I'd like to debug it myself so I can provide a better feed-back. (Also the camera parameters). And do you use BODY_25 (default) or COCO model?

PS: I saw your pull request, I'll take a deeper look to the code, but your fix looks good so thank you for it!

**Comment by gineshidalgo99** [comment]
Sorry for the huge delay. So I tried your videos and images. The problem is the ffmpeg conversion, OpenPose behaves 100% the same for videos and images (assuming they provide the exact same raw information, which is not the case here).

If you use lossless quality PNGs, and then turn those frames into lossy video format, then the quality will drop depending on how extreme the compression is.

My test: I had the original PNG images, turned them into video. The results are no longer the same. However, given that video, I turned it into PNG images. These last 2 (video and images generated from the video) provide 100% the exact same results.

Thanks for sharing the images and video! That was useful for me to test it!
